app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -101,6 +110,45 @@ class MessageActivity : AppCompatActivity() {
             }
         }
 
+        save.setOnClickListener {
+            try {
+                val inputManager: InputMethodManager = this.getSystemService(Context.INPUT_METHOD_SERVICE)
+                        as InputMethodManager
+                inputManager.hideSoftInputFromWindow(content.windowToken, InputMethodManager.SHOW_FORCED)
+
+                val sdf = SimpleDateFormat("dd-MMM-yyyy-hh_mm_aaa")
+                val fileName = "badge-magic-" + sdf.format(Calendar.getInstance().time).toString()
+                val os = FileOutputStream(fileName, true)
+                var text = "null"
+                var drawableID = "null"
+                var flashSelected = false
+                var marqueeSelected = false
+
+                if (content.text.isNotBlank())
+                    text = content.text.toString()
+                else if (drawableRecyclerAdapter.selectedPosition >= 0)
+                    drawableID = drawableRecyclerAdapter.selectedPosition.toString()
+
+                val selectedSpeed = speed.selectedItem.toString()
+                val selectedMode = mode.selectedItem.toString()
+
+                when {
+                    flash.isChecked -> flashSelected = true
+                    marquee.isChecked -> marqueeSelected = true
+                }
+
+                moshiConverter.toStream((text + drawableID + selectedSpeed + selectedMode + flashSelected.toString() + marqueeSelected.toString()),os)
```

Comment:
Why drawableSelected or Text. Rather lets store the hexadecimal strings itself. That would give us the option to load the strings directly on the preview

Reply:
I'll look at it